packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1466,11 +1478,126 @@ export class ContainerRuntime extends EventEmitter
                 builder.addWithStats(key, contextSummary);
             }));
 
+        const deletedDataStores = this.runGarbageCollection();
+        builder.addBlob(".deletedDataStores", JSON.stringify(deletedDataStores));
+
         this.serializeContainerBlobs(builder);
         const summary = builder.getSummaryTree();
         return { ...summary, id: "" };
     }
 
+    public runGarbageCollection(): string[] {
+        // Maintains a map of dataStoreId to an IReferences object. The IReferences object keeps track of whether
+        // the data store has been visited (to detect loops) and the fluid objects in it that are referenced.
+        const dataStoreReferencesMap: Map<string, IReferences> = new Map();
+
+        // A stack of routes that are referenced through handles in DDSes.
+        const referencedRoutesStack: string[] = [];
+
+        // Start with the root data stores. Add all the fluid objects referenced by the root data stores in to the
+        // referencedObjectsStack.
+        this.contexts.forEach((value: FluidDataStoreContext) => {
+            if (value.isRootStore) {
+                // Ignore the scheduler.
+                if (value.id === taskSchedulerId) {
```

Comment:
why task scheduler is special?

Reply:
I guess this is because of my lack of understanding of the scheduler. I thought we cannot put references to other fluid objects in the scheduler. Is that understanding not correct?

Reply:
Even if it's the case, it does not matter - it will return empty set. We should not make any exclusions like that, this allows system to be flexible, i.e. there is no need to put restrictions around something that may want not to have such restrictions in the future